src/Angor/Avalonia/Angor.Contexts.Wallet/Infrastructure/Impl/WalletAppService.cs

```diff
@@ -182,4 +198,29 @@ public string GenerateRandomSeedwords()
         var mnemonic = new Mnemonic(Wordlist.English, WordCount.Twelve);
         return mnemonic.ToString();
     }
+
+    private async Task<Result<long>> CalculateTransactionFee(SendInfo sendInfo, AccountInfo accountInfo, WalletWords walletWords, DomainFeeRate feeRate)
```

Comment:
you are only doing this for the fee size? I suggest keeping the signedTransaction no? so that you wont need to build it again (not that it matter on desktop is it probably pretty fast)

Reply:
There is a property on sendInfo you could just store the trx there

Reply:
I don't think caching will help here, and In any case to get real fee estimation you need to create the signed trx and know its exact size, so it is hard to get the fee only (accurately) without building the entire signed trx.
If this is a perf issue we can revist